### ISSUE TITLE
inception_train without gpu

### ISSUE THREAD

**Issue by kdsoo** [tensorflow/models#587]
I find that the inception training example only runs on systems with gpgpu and num_gpus is forced to be set more than 1 gpu.

Is the inception model not feasible to be trained only with cpus? Because so far I did transfer learning with inception model and did not require gpu, but now I am trying to intake inception transfer learning in tensorflow serving which requires a single piece of exported checkpoints file and meta graph file instead.

As far as I tried, inception_train with no num_gpus option specified just get trained well but trouble happens on launching model server which seems to lookup GPU:0 device when loading model checkpoints but sadly my system doesn't have cuda compatible gpu.

If there already exists some pointer for systems without gpus, I will happily follow that.

**Comment by drpngx** [comment]
Inception models are pretty expensive. I am not sure it makes sense to try without GPUs.

**Comment by TRGNN** [comment]
You should try :

`export CUDA_VISIBLE_DEVICES=""`

**Comment by kdsoo** [comment]
@drpngx You are absolutely right.
What I tried since was only querying image against pretrained inception model v3 and transfer learning which performs last layer of the process and fortunately those didn't require expensive computation.
I just hope to test inception model training on non-gpu docker instance running on intel mobile platform.
Not for commercial service so not time critical. just test purpose :)

**Comment by kdsoo** [comment]
@TRGNN unfortunately it didn't go well.

1) I tried export CUDA_VISIBLE_DEVICES='' in shell and ran inception train and model server ==> no luck
2) tried CUDA_VISIBLE_DEVICES='' bazel-bin/.... (train and model server as well) ==> no luck

here is the error log.

root@2b2844638a74:/serving# CUDA_VISIBLE_DEVICES='' bazel-bin/tensorflow_serving/model_servers/tensorflow_model_server --port=9000 --model_name=inception --model_base_path=/tmp/flower_train
I tensorflow_serving/model_servers/main.cc:125] Building single TensorFlow model file config:  model_name: inception model_base_path: /tmp/flower_train
I tensorflow_serving/core/basic_manager.cc:46] Creating InlineExecutor for BasicManager.
I tensorflow_serving/model_servers/server_core.cc:219] Adding/updating models.
I tensorflow_serving/model_servers/server_core.cc:252]  (Re-)adding model: inception
I tensorflow_serving/core/basic_manager.cc:49] Creating ThreadPoolExecutor for BasicManager with num_load_unload_threads: 16
I tensorflow_serving/sources/storage_path/file_system_storage_path_source.cc:252] File-system polling update: Servable:{name: inception version: 157586}; Servable path: /tmp/flower_train/00157586; Polling frequency: 30
I tensorflow_serving/core/basic_manager.cc:646] Successfully reserved resources to load servable {name: inception version: 157586}
I tensorflow_serving/core/loader_harness.cc:70] Approving load for servable version {name: inception version: 157586}
I tensorflow_serving/core/loader_harness.cc:85] Loading servable version {name: inception version: 157586}
I external/org_tensorflow/tensorflow/contrib/session_bundle/session_bundle.cc:142] Attempting to load a SessionBundle from: /tmp/flower_train/00157586
I external/org_tensorflow/tensorflow/contrib/session_bundle/session_bundle.cc:143] Using RunOptions: 
I external/org_tensorflow/tensorflow/contrib/session_bundle/session_bundle.cc:106] Running restore op for SessionBundle
E tensorflow_serving/core/loader_harness.cc:104] Servable: {name: inception version: 157586} load failure: Invalid argument: Cannot assign a device to node 'tower_0/gradients/tower_0/conv0/BatchNorm/moments/sufficient_statistics/mean_ss_grad/Fill': Could not satisfy explicit device specification '/device:GPU:0' because no devices matching that specification are registered in this process; available devices: /job:localhost/replica:0/task:0/cpu:0
	 [[Node: tower_0/gradients/tower_0/conv0/BatchNorm/moments/sufficient_statistics/mean_ss_grad/Fill = Fill[T=DT_INT32, _output_shapes=[[3]], _device="/device:GPU:0"](tower_0/gradients/tower_0/conv0/BatchNorm/moments/sufficient_statistics/mean_ss_grad/Shape_1, tower_0/gradients/tower_0/conv0/BatchNorm/moments/sufficient_statistics/mean_ss_grad/Fill/value)]]

**Comment by drpngx** [comment]
Can you replace with the gpu placement with gpu?

```
available devices: /job:localhost/replica:0/task:0/cpu:0
```

**Comment by kdsoo** [comment]
@drpngx I took a look at the meta graph file extracted by inception train, and ended up to make a cpu version of inception_train.py.
inception_train.py is forcing to use gpu resource even I give a notice by setting environment variable CUDA_VISIBLE_DEVICES=''  that my system doesn't have any cuda compatible gpu. 
Not a surprise going through the inception_train code, It enforces to set using gpu in the first place.(no escape)
Anyway, I'll try to contribute if I finish making cpu only version. Even I may be the only person who need that.

**Comment by drpngx** [comment]
Great! If nothing else, that could be useful for debugging.
